framework/include/utils/RankTwoTensor.h

```diff
@@ -502,6 +548,9 @@ class RankTwoTensorTempl : public TensorValue<T>
   /// RankTwoTensorTempl<T> from outer product of vectors
   void vectorOuterProduct(const TypeVector<T> &, const TypeVector<T> &);
```

Comment:
Perhaps this should also be a named constructor.

Reply:
Ha, yeah, for sure. I just made that change and got bit hard. The old syntax of calling it

```
mytensor.vectorOuterProduct(v, v);
```

 is still totally valid, but now simply discards the return value and does nothing :-D

Fortunately with C++17 we now have the `[[nodiscard]]` attribute available. So I slapped that on to make this at least give a compiler warning. Would be nice if it was an error though!

Reply:
Good job C++17 :+1: 

I thought we promote compiler warnings to errors all the time. Maybe that have been changed at some point?

Reply:
Alternatively, we could deprecate the current version of `vectorOuterProduct` in favor of a new named constructor called something like `initializeFromVectorOuterProduct` so that it doesn't break anything, and we could deal with the deprecation in a later PR.

Reply:
Yeah, i'd probably go with something shorter rather than longer (the vector part should be clear from the arguments). Let's see if we get failing app tests. Might be worth checking if we have any apps that don't get compiled with `-Werror`